tests/components/calendar/test_trigger.py

```diff
@@ -99,6 +105,13 @@ async def fire_time(self, trigger_time: datetime.datetime) -> None:
 
     async def fire_until(self, end: datetime.timedelta) -> None:
         """Simulate the passage of time by firing alarms until the time is reached."""
+        current_time = dt_util.as_utc(self.freezer())
```

Comment:
The motivation for this is the test harness was incorrectly comparing all day events. This was just a test improvement, and is here as we've added more test coverage to more exhaustively test different types of calendar payloads